examples/autoignition-mpi.py

```diff
@@ -0,0 +1,229 @@
+    i_ox = cantera_soln.species_index("O2")
+    i_di = cantera_soln.species_index("N2")
+    x = np.zeros(nspecies)
+    x[i_fu] = (ox_di_ratio*equiv_ratio)/(stoich_ratio+ox_di_ratio*equiv_ratio)
+    x[i_ox] = stoich_ratio*x[i_fu]/equiv_ratio
+    x[i_di] = (1.0-ox_di_ratio)*x[i_ox]/ox_di_ratio
+    # one_atm = cantera.one_atm
+    one_atm = 101325.0
+
+    print(f"Input state (T,P,X) = ({init_temperature}, {one_atm}, {x}")
+    cantera_soln.TPX = init_temperature, one_atm, x
+    can_t, can_rho, can_y = cantera_soln.TDY
+    can_p = cantera_soln.P
+
+    casename = "autoignition"
+    pyrometheus_mechanism = pyro.get_thermochem_class(cantera_soln)(actx.np)
+    eos = PyrometheusMixture(pyrometheus_mechanism, tguess=init_temperature)
+
+    print(f"Cantera state (rho,T,P,Y) = ({can_rho}, {can_t}, {can_p}, {can_y}")
+    initializer = MixtureInitializer(dim=dim, nspecies=nspecies,
+                                     pressure=can_p, temperature=can_t,
+                                     massfractions=can_y, velocity=velocity)
+
+    my_boundary = AdiabaticSlipBoundary()
+    boundaries = {BTAG_ALL: my_boundary}
+    current_state = initializer(eos=eos, x_vec=nodes, t=0)
+
+    # Inspection at physics debugging time
+    #    cv = split_conserved(dim, current_state)
+    #    print(f"Initial CV rho: {cv.mass}")
+    #    print(f"Initial CV rhoE: {cv.energy}")
+    #    print(f"Initial CV rhoV: {cv.momentum}")
+    #    print(f"Initial CV rhoY: {cv.mass_fractions}")
+    #    print(f"Initial Y: {cv.mass_fractions / cv.mass}")
+    #    print(f"Initial DV pressure: {eos.pressure(cv)}")
+    #    print(f"Initial DV temperature: {eos.temperature(cv)}")
+
+    visualizer = make_visualizer(discr, discr.order + 3
+                                 if discr.dim == 2 else discr.order)
+    initname = initializer.__class__.__name__
+    eosname = eos.__class__.__name__
+    init_message = make_init_message(dim=dim, order=order,
+                                     nelements=local_nelements,
+                                     global_nelements=global_nelements,
+                                     dt=current_dt, t_final=t_final, nstatus=nstatus,
+                                     nviz=nviz, cfl=current_cfl,
+                                     constant_cfl=constant_cfl, initname=initname,
+                                     eosname=eosname, casename=casename)
+    cantera_soln.equilibrate("UV")
+    can_eq_t, can_eq_rho, can_eq_y = cantera_soln.TDY
+    can_eq_p = cantera_soln.P
+    #    can_eq_e = cantera_soln.int_energy_mass
+    #    can_eq_k = cantera_soln.forward_rate_constants
+    #    can_eq_c = cantera_soln.concentrations
+    #    can_eq_r = cantera_soln.net_rates_of_progress
+    #    can_eq_omega = cantera_soln.net_production_rates
+    if rank == 0:
+        logger.info(init_message)
+        logger.info(f"Expected (p,T,rho, y) = ({can_eq_p}, {can_eq_t},"
+                    f" {can_eq_rho}, {can_eq_y})")
```

Comment:
Read up on `f"{var_name=}"` to avoid having to print labels.

Reply:
What if I want the labels, though?

Reply:
I think this is what you are looking for here, but I am not sure that this is better than the original message. What do you think? (9db65b7)

Reply:
LGTM, but feel free to revert if you prefer the previous style.